### ISSUE TITLE
Does DataLoader(shuffle=True) really shuffle DBpedia dataset correctly?

### ISSUE THREAD

**Issue by fujidaiti** [pytorch/text#2197]
According to [the docs][1], DBpedia dataset has 14 classes (labels) and 40000 texts for each class. Then, if I create batches using `DataLoader(shuffle=True)` as follows:

```python
import torchtext.datasets as d
from torch.utils.data.dataloader import DataLoader

train = DataLoader(
    d.DBpedia(split="train", root=".cache"),
    batch_size=10000,
    shuffle=True,
)
```

the labels should be uniformly distributed in each batch. But in practice, it seems that only a few labels are in each batch.

```python
for labels, texts in train:
    print(len(set(labels.tolist())))
```
The output of the above code is:
```
1
1
1
2
2
2
2
3
3
3
3
4
4
3
3
.
.
.
```

How can I fix this? Or is my implementation wrong?

P.S.
Interactive code is available on [GoogleColab][2]

  [1]: https://pytorch.org/text/stable/datasets.html#dbpedia
  [2]: https://colab.research.google.com/drive/10524PcR3_spf3fAh37hNbXdLeRVD6Sog?usp=sharing
